### ISSUE TITLE
CSM Utility Daemon is unable to reconnect to CSM Master Daemon

### ISSUE THREAD

**Issue by besawn** [IBM/CAST#128]
The problem was originally detected via a failure of an LSF daemon to communicate with the local CSM Utility daemon. Attempts to manually run /opt/ibm/csm/bin/csm_node_resources_query_all (or any other CSM command line program) on the utility node in question fail with the following message:
```
$ /opt/ibm/csm/bin/csm_node_resources_query_all

[csmapi][error] csmlib: Failed to connect to CSM_SSOCKET=/run/csmd.sock; retried 9 times.
csm_net_unix_Connect: No such file or directory
[csmapi][error] csm_net_unix_Connect() failed: /run/csmd.sock
[csmapi][error] /home/ppsbld/workspace/PUBLIC_CAST_V1.1.1_ppc64LE_RH7.5_ProdBuild/csmi/src/wm/cmd/node_resources_query_all.c-166:
[csmapi][error]   csm_init_lib rc= 2, Initialization failed. Success is required to be able to communicate between library and daemon. Are the daemons running?
```

The message above is considered a normal response when the local CSM daemon is unable to establish a connection with the CSM Master or Aggregator daemon. When there is a daemon-to-daemon communication failure, CSM intentionally stops listening for local client connections until the daemon-to-daemon communication is restored.

Through examination of the CSM Utility and Master daemon logs, the following behavior is observed:
- The Utility daemon is started ~10 minutes before the problem occurs.
- The Utility daemon is able to successfully connect to the Master daemon and process several message over several minutes.
- An unknown error occurs, causing the Utility daemon to detect the connection with the Master daemon is down via a failed heartbeat.
- The Utility daemon closes the connection to the Master daemon.
- The Utility daemon stops accepting local client requests, as designed. Local client API calls all fail with the message above.
- The Utiltity daemon then enters a retry loop, where it attempts to reconnect to the Master daemon, ~ every 50 seconds. However, it is never able to successfully reconnect, despite retrying for over 2 hours.
- Finally, a restart of just the Utility daemon allows it to successfully reconnect to the Master and normal operation is restored.

During each retry cycle, the following pattern is observed:
- The Utility daemon creates a new connection to the Master daemon and sends a version message.
- The Master daemon accepts the connection and processes the version message.
- Somewhere between the two daemons, the response from the version message from the Master to Utility daemon is lost, causing the Utility daemon to detect the connection as down and close it.

Open questions:
1.) What was the initial error that triggered this behavior?
2.) What is causing the connection retry handshake to fail, despite repeated attempts to bring the connection back?

The installed version of CSM is:
```
ibm-csm-core-1.1.1-163
Version: 1.1.1; Build: fc7ceab25b; Date: Tue May 22 14:51:00 2018 -0400;
```

**Comment by fpizzano** [comment]
Adding @miidon @verolero86 and @mattaezell

**Comment by fpizzano** [comment]
Logs are located @ /ugadmin/csm/CAST/128

**Comment by mew2057** [comment]
@lasch  @fpizzano  Was this fixed in #240?

**Comment by fpizzano** [comment]
Closing after Veronica and Don agreement. We will open a new issue if reoccurred.
